test_programs/execution_success/array_dedup_regression/src/main.nr

```diff
@@ -0,0 +1,21 @@
+unconstrained fn main(x: u32) {
+    let a1 = [1, 2, 3, 4, 5];
+
+    for i in 0..5 {
+        let mut a2 = [1, 2, 3, 4, 5];
+        a2[x + i] = 128;
+        println(a2);
+
+        if i != 0 {
+            assert(a2[x + i - 1] != 128);
+        }
+    }
+
+    // Can't use `== [1, 2, 3, 4, 5]` here, that make_array may get
+    // deduplicated to equal a1 in the bugged version
```

Comment:
This still occurs in this PR?

Reply:
No but in some versions of the buggy version it'd get deduplicated to the mutated version. To avoid confusion in that version I just checked each index manually.